### PR TITLE
Added inorder.cpp file for inorder tree traversal

### PR DESCRIPTION
**Pull request by Chaitanya134**
#### Description of Change
Implemented inorder tree traversal

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by Chaitanya134** [comment]
Hey, @mishraabhinn Can you please review this pr for hacktoberfest?

**Comment by Chaitanya134** [comment]
@Panquesito7 Can you please review this?

File: data_structures/inorder.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/**
+ * @file
+ * @brief [Inorder Tree Traveral](https://www.javatpoint.com/inorder-traversal)
+ * @author [Chaitanya Giri](https://github.com/Chaitanya134)
+ */
+
+#include <iostream>  /// for IO operations
+#include <cassert>   /// for assert
+#include <queue>     /// for storing node in level order traversal
+using namespace std;
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:10]: This is discouraged. Please strictly use `std::` for each member instead. You can use `using std::cout;`, `using std::cin;`, and so on for all `std::` members.

File: data_structures/inorder.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    */
+    void inorderTraversal(Node *node) {
+        if (node == NULL) return;       // stop traversal if node is NULL
+
+        inorderTraversal(node->left);   // recursively traverse current node's left subtree
+        cout << node->data << " ";      // print the current node's data
+        inorderTraversal(node->right);  // recursively traverse current node's right subtree
+    }
+};
+
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:81]: Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord). 😃

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: data_structures/inorder.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ * @brief [Inorder Tree Traveral](https://www.javatpoint.com/inorder-traversal)
+ * @author [Chaitanya Giri](https://github.com/Chaitanya134)
+ */
+
+#include <iostream>  /// for IO operations
+#include <cassert>   /// for assert
+#include <queue>     /// for storing node in level order traversal
+#include <sstream>   /// for converting expected output to a string
+
+class Node {
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:12]:
```suggestion
/**
 * @namespace
 * @brief Data structures
 */
namespace data_structures {
class Node {
```

File: data_structures/inorder.cpp [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        inorderTraversal(node->right);  // recursively traverse current node's right subtree
+    }
+};
+
+/**
+* @brief [Self-tests for inorder tree traversal]
+* @returns void
+*/
+void test() {
+    BinaryTree tree;    // create a tree
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:87]:
```suggestion
    data_structures::BinaryTree tree;    // create a tree
```

File: data_structures/inorder.cpp [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    // Testing inorder traversal
+    test();
+
+    BinaryTree tree;    // create a tree
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:152]:
```suggestion
    data_structures::BinaryTree tree;    // create a tree
```

File: data_structures/inorder.cpp [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    Node *root = NULL;  // root node
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:153]:
```suggestion
    data_structures::Node *root = nullptr;  // root node
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: data_structures/inorder.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (Chaitanya134) [reply data_structures/inorder.cpp:12]: Can you suggest the correct way to write a class description?

[the remark this answers is not in the document]
Comment (Panquesito7) [reply data_structures/inorder.cpp:25]: You can use something similar to this, or anything else you'd like; it doesn't have to be long. 🙂

```cpp
/**
 * @brief Class that includes the required functions
 * to make the code work properly.
 */
```

[the remark this answers is not in the document]
Comment (Chaitanya134) [reply data_structures/inorder.cpp:25]: Okay got it. My recent commit has class description in it.

File: data_structures/inorder.cpp [hunk 7]
```diff
@@ -0,0 +1,181 @@
+/**
+ * @file
+ * @brief [Inorder Tree Traveral](https://www.javatpoint.com/inorder-traversal)
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:3]: Please add a detailed description (you can take as a reference any source link, such as Wikipedia or the current one).

File: data_structures/inorder.cpp [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+        inorderTraversal(node->left);   // recursively traverse current node's left subtree
+        std::cout << node->data << " ";      // print the current node's data
+        inorderTraversal(node->right);  // recursively traverse current node's right subtree
+    }
+};
+} // namespace data_structures
+
+/**
+* @brief [Self-tests for inorder tree traversal]
```
Comment (Panquesito7) [inline data_structures/inorder.cpp:104]: There's no need for the brackets. That's usually used only for Markdown links.
```suggestion
* @brief Self-tests for inorder tree traversal
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**